frontend/src/metabase/components/form/FormField/FormField.tsx

```diff
@@ -93,8 +93,8 @@ function FormField({
     ...props,
   };
 
-  const shouldHideError = !visited || active;
-  const error = shouldHideError ? undefined : errorProp;
+  const shouldShowError = visited && !active;
+  const error = !shouldShowError ? undefined : errorProp;
```

Comment:
Simplify logic a bit. I assume readers would be more interested in how the error is displayed not how we're hiding the error.